compiler/executable/match_/planner/plan.rs

```diff
@@ -941,29 +941,42 @@ impl PartialCostPlan {
     }
 
     fn determine_joinability(&self, graph: &Graph<'_>, pattern: PatternVertexId) -> Option<VariableVertexId> {
-        let mut updated_join_var: Option<VariableVertexId> = None;
-        if let Some(prev_constraint) = self.ongoing_step.iter().next() {
+        if let Some(&prev_pattern) = self.ongoing_step.iter().next() {
+            // We only join constraint patterns, so let's extract constraints
+            let prev_planner = &graph.elements[&VertexId::Pattern(prev_pattern)];
+            let PlannerVertex::Constraint(prev_constraint) = prev_planner else {
+                return None;
+            };
```

Comment:
If you remove semicolons in the else blocks of let-else, rustfmt will happily keep them on one line ;)
```suggestion
            let PlannerVertex::Constraint(prev_constraint) = prev_planner else { return None };
```